store/bsdb/prefix_test.go

```diff
@@ -0,0 +1,227 @@
+			WithArgs(common.HexToHash("2")).
+			WillReturnRows(
+				sqlmock.NewRows([]string{"object_id", "object_name"}).
+					AddRow(common.HexToHash("2"), "obj2"))
+	}
+
+	res, err := s.ListObjects(bucketName, "", prefix, maxKeys)
+	assert.Nil(t, err)
+	assert.Equal(t, expectedObjects, res)
+}
+
+func TestBsDBImpl_ListObjectsWithPrefixQuery(t *testing.T) {
+	prefix := "testPrefix/testQuery"
+	maxKeys := 2
+	expectedObjects := []*ListObjectsResult{
+		{PathName: "testPath2/obj2", ResultType: "object", Object: &Object{ObjectName: "obj2", ObjectID: common.HexToHash("2")}},
+	}
+
+	s, mock := setupDB(t)
+
+	// Split the prefix into the path and the query for the LIKE clause.
+	splitIdx := strings.LastIndex(prefix, "/")
+	pathName := prefix[:splitIdx+1]
+	nameQuery := prefix[splitIdx+1:] + "%"
+
+	// Modify the expected SQL to include the name LIKE clause
+	expectedSQL1 := "SELECT * FROM `slash_prefix_tree_nodes_31` WHERE bucket_name = ? AND path_name = ? AND name like ? ORDER BY full_name LIMIT 3"
+	mock.ExpectQuery(expectedSQL1).
+		WithArgs(bucketName, pathName, nameQuery).
+		WillReturnRows(
+			sqlmock.NewRows([]string{"path_name", "full_name", "name", "is_object", "is_folder", "bucket_name", "object_id", "object_name"}).
+				AddRow("testPath2", "testPath2/obj2", "obj2", true, false, bucketName, common.HexToHash("2"), "obj2"))
+
+	// Set expectation for the object details query on the respective objects_xx table
+	for i := 0; i < 64; i++ {
+		mock.ExpectQuery(fmt.Sprintf("SELECT * FROM `%s` WHERE object_id in (?)", GetObjectsTableNameByShardNumber(i))).
+			WithArgs(common.HexToHash("2")).
+			WillReturnRows(
+				sqlmock.NewRows([]string{"object_id", "object_name"}).
+					AddRow(common.HexToHash("2"), "obj2"))
+	}
+
+	res, err := s.ListObjects(bucketName, "", prefix, maxKeys)
+	assert.Nil(t, err)
+	assert.Equal(t, expectedObjects, res)
+}
+
+func TestBsDBImpl_ListObjectsWithAllConditions(t *testing.T) {
+	prefix := "testPath/testQuery"
+	maxKeys := 2
+	continuationToken := "testToken"
+	expectedObjects := []*ListObjectsResult{
+		{PathName: "testPath2/obj2", ResultType: "object", Object: &Object{ObjectName: "obj2", ObjectID: common.HexToHash("2")}},
+	}
+
+	s, mock := setupDB(t)
+
+	// Split the prefix into the path and the query for the LIKE clause.
+	splitIdx := strings.LastIndex(prefix, "/")
+	pathName := prefix[:splitIdx+1]
+	nameQuery := prefix[splitIdx+1:] + "%"
+
+	expectedSQL := "SELECT * FROM `slash_prefix_tree_nodes_31` WHERE bucket_name = ? AND path_name = ? AND name like ? AND full_name >= ? ORDER BY full_name LIMIT 3"
+	mock.ExpectQuery(expectedSQL).
+		WithArgs(bucketName, pathName, nameQuery, continuationToken).
+		WillReturnRows(
+			sqlmock.NewRows([]string{"path_name", "full_name", "name", "is_object", "is_folder", "bucket_name", "object_id", "object_name"}).
+				AddRow("testPath2", "testPath2/obj2", "obj2", true, false, bucketName, common.HexToHash("2"), "obj2"))
+
+	for i := 0; i < 64; i++ {
```

Comment:
do we have a const indicating the total number of sharding tables

Reply:
fixed